### PR TITLE
feat: [SRM-13326]: Handle Simple SLO with same identifier in composite SLO

### PR DESCRIPTION
**Pull request by deepesh-ui**
### Summary

Handle Simple SLO with same identifier in Account 

#### Screenshots

Screenshot 2022-12-12 at 4 38 18 PM

Screenshot 2022-12-12 at 4 37 59 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx [hunk 1]
```diff
@@ -157,7 +163,14 @@ interface RenderCheckBoxesInterface {
 
 export const RenderCheckBoxes = ({ row, selectedSlos, setSelectedSlos }: RenderCheckBoxesInterface) => {
   const sloData = row.original
-  const isChecked = Boolean([...selectedSlos].find(item => item.name === sloData.name))
+  // const isChecked = Boolean([...selectedSlos].find(item => item.name === sloData.name))
+  const isChecked = Boolean(
+    [...selectedSlos].find(
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx:168]: why cant we simply to `selectedSlos?.find`

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx [hunk 2]
```diff
@@ -157,7 +163,14 @@ interface RenderCheckBoxesInterface {
...
+      item =>
+        `${item.sloIdentifier}.${item.projectParams?.orgIdentifier}.${item.projectParams?.orgIdentifier}` ===
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx:170]: destructure fields from `projectParams`

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx [hunk 3]
```diff
@@ -145,7 +145,13 @@ export const onSelectCheckBox = (
     clonedSelectedSlos.push(slo)
     setSelectedSlos(clonedSelectedSlos)
   } else {
-    setSelectedSlos(clonedSelectedSlos.filter(item => item.name !== slo.name))
+    setSelectedSlos(
+      clonedSelectedSlos.filter(
+        item =>
+          `${item.sloIdentifier}.${item.projectParams?.orgIdentifier}.${item.projectParams?.orgIdentifier}` !==
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx:151]: please destructure fields.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 4]
```diff
@@ -92,7 +92,10 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
         return {
           ...sloDetail,
           ...(dashboardWidgetsResponse?.data?.content?.find(
-            item => item.sloIdentifier === sloDetail.serviceLevelObjectiveRef
+            // item => item.sloIdentifier === sloDetail.serviceLevelObjectiveRef
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:95]: commented code.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 5]
```diff
@@ -92,7 +92,10 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
...
+            item =>
+              `${item.sloIdentifier}.${item.projectParams?.orgIdentifier}.${item.projectParams?.projectIdentifier}` ===
```
Comment (harshilbits) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:97]: These statements are repeated at multiple places. . Please reuse them.

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx [hunk 6]
```diff
@@ -91,9 +95,13 @@ export const AddSLOs = (props: AddSLOsProp): JSX.Element => {
       const updatedSLODetails = formikProps?.values?.serviceLevelObjectivesDetails?.map(sloDetail => {
         return {
           ...sloDetail,
-          ...(dashboardWidgetsResponse?.data?.content?.find(
-            item => item.sloIdentifier === sloDetail.serviceLevelObjectiveRef
-          ) as SLOHealthListView)
+          ...(isAccountLevel
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx:98]: Can we move this to a utility function? It is hard to understand here

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx [hunk 7]
```diff
@@ -138,31 +139,45 @@ export const onSelectCheckBox = (
...
   setSelectedSlos: React.Dispatch<React.SetStateAction<SLOHealthListView[]>>
+  isAccountLevel?: boolean
 }
 
-export const RenderCheckBoxes = ({ row, selectedSlos, setSelectedSlos }: RenderCheckBoxesInterface) => {
+export const RenderCheckBoxes = ({ row, selectedSlos, setSelectedSlos, isAccountLevel }: RenderCheckBoxesInterface) => {
   const sloData = row.original
-  const isChecked = Boolean([...selectedSlos].find(item => item.name === sloData.name))
+  const isChecked = isAccountLevel
+    ? Boolean(
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx:170]: use `Array.some` instead

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx [hunk 8]
```diff
@@ -138,31 +139,45 @@ export const onSelectCheckBox = (
...
+        selectedSlos.find(
+          item => getSLOIdentifierWithOrgAndProject(item) === getSLOIdentifierWithOrgAndProject(sloData)
+        )
+      )
+    : Boolean(selectedSlos.find(item => item.name === sloData.name))
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx:175]: use `Array.some`

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/CVCreateSLOV2.utils.ts [hunk 9]
```diff
@@ -24,7 +24,10 @@ export const filterServiceLevelObjectivesDetailsFromSLOObjective = (
 ): ServiceLevelObjectiveDetailsDTO[] =>
   serviceLevelObjectivesDetails
     ? serviceLevelObjectivesDetails.map(
-        sloDetail => pick(sloDetail, [...serviceLevelObjectiveKeys]) as ServiceLevelObjectiveDetailsDTO
+        sloDetail =>
+          pick({ ...sloDetail, ...sloDetail?.projectParams }, [
```
[the remark this answers is not in the document]
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/CVCreateSLOV2.utils.ts]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx (hunk 4 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx (hunk 5 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/AddSLOs.tsx]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx [hunk 10]
```diff
@@ -157,7 +163,14 @@ interface RenderCheckBoxesInterface {
 
 export const RenderCheckBoxes = ({ row, selectedSlos, setSelectedSlos }: RenderCheckBoxesInterface) => {
   const sloData = row.original
-  const isChecked = Boolean([...selectedSlos].find(item => item.name === sloData.name))
+  // const isChecked = Boolean([...selectedSlos].find(item => item.name === sloData.name))
```
[the remark this answers is not in the document]
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx (hunk 8 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx]: done

File: src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx (hunk 7 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/slos/components/CVCreateSLOV2/components/CreateCompositeSloForm/components/AddSlos/components/SLOList.utils.tsx]: done

**Review by praneshg239 — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**